DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/DateShiftProcessor.cs

```diff
@@ -0,0 +1,86 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class DateShiftProcessor : IAnonymizerProcessor
+    {
+        private readonly DateShiftFunction _dateShiftFunction;
+        private readonly DateShiftScope _dateShiftScope = DateShiftScope.SopInstance;
+
+        public DateShiftProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
```

Comment:
>settingFactory ??= new AnonymizerSettingsFactory(); [](http://example.com/codeflow?start=12&length=51)

look like all the processors that needs settingFactory will create the factory if it is null, so why not simply pass it forcefully instead of creating it when it is null?

Reply:
Make sense!

Reply:
Not all processors needs settingFactory such as remove, substitute, refreshUID.